core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html

```diff
@@ -0,0 +1,174 @@
+           focus-on="<[::LABEL_FOR_CLEARING_FOCUS]>">
+          <img ng-src="<[getStaticImageUrl('/logo/288x128_logo_white.png')]>" class="oppia-logo" ng-class="windowIsNarrow ? 'oppia-logo-small' : 'oppia-logo-wide'">
+        </a>
+        <!-- This is needed for the correct image to appear when an exploration is shared using G+. -->
+        <a style="display: none;">
+          <img ng-src="<[getStaticImageUrl('/logo/288x128_logo_mint.png')]>" itemprop="image">
+        </a>
+      </div>
+
+      <div ng-cloak class="navbar-header pull-right" ng-if="NAV_MODE !== 'signup'">
+        <ul class="nav oppia-navbar-nav oppia-navbar-profile">
+          {% if username %}
+            <li class="dropdown pull-right protractor-test-profile-dropdown">
+              <a class="dropdown-toggle oppia-navbar-dropdown-toggle" data-toggle="dropdown"
+                 ng-mouseover="onMouseoverProfilePictureOrDropdown($event)"
+                 ng-mouseleave="onMouseoutProfilePictureOrDropdown($event)">
+                <div class="oppia-navbar-profile-picture-container" ng-cloak>
+                  {% if profile_picture_data_url %}
+                    <img src="{{profile_picture_data_url}}" class="oppia-navbar-profile-picture img-circle">
+                    <span class="caret" style="margin-top: 10px;"></span>
+                  {% else %}
+                    <i class="material-icons md-40" style="margin-top: -1px;">&#xE853;</i>
+                    <span class="caret" style="margin-top: -26px;"></span>
+                  {% endif %}
+
+                  <div class="oppia-navbar-dashboard-indicator ng-cloak" ng-if="numUnseenNotifications > 0">
+                    <span class="oppia-navbar-dashboard-indicator-text">
+                      <[numUnseenNotifications]>
+                    </span>
+                  </div>
+
+                  {% if is_admin or is_moderator %}
+                    <div class="oppia-navbar-role-indicator">
+                      {% if is_admin %}
+                        <!-- "right: 4px;" is necessary here but not in moderator to prevent 'A' from appearing off-center because 'A' is slightly thinner than 'M' in this font -->
+                        <span class="oppia-navbar-role-text" style="right: 4px;">A</span>
+                      {% elif is_moderator %}
+                        <span class="oppia-navbar-role-text">M</span>
+                      {% endif %}
+                    </div>
+                  {% endif %}
+                </div>
+              </a>
+
+              <ul class="dropdown-menu ng-cloak oppia-navbar-dropdown" role="menu"
+                  ng-mouseover="onMouseoverProfilePictureOrDropdown($event)"
+                  ng-mouseleave="onMouseoutProfilePictureOrDropdown($event)"
+                  ng-show="profileDropdownIsActive">
+                <li>
+                  <a ng-click="onMouseoutProfilePictureOrDropdown($event)" href="/profile/{{username}}"
```

Comment:
This is one minus of not using Jinja: easy interpolation of username. Then again,we do have ``UrlInterpolationService``, so maybe not a big deal.